apps/mark-scan/frontend/src/voter_flow.tsx

```diff
@@ -79,19 +80,8 @@ export function VoterFlow(props: VoterFlowProps): React.ReactNode {
     );
   }
 
-  if (stateMachineState === 'waiting_for_ballot_reinsertion') {
-    return <WaitingForBallotReinsertionBallotScreen />;
-  }
-
-  if (
-    stateMachineState === 'loading_reinserted_ballot' ||
-    stateMachineState === 'validating_reinserted_ballot'
-  ) {
-    return <LoadingReinsertedBallotScreen />;
-  }
-
-  if (stateMachineState === 'reinserted_invalid_ballot') {
-    return <ReinsertedInvalidBallotScreen />;
+  if (isBallotReinsertionState(stateMachineState)) {
```

Comment:
Thanks for subcategorizing these flows. Helps with readability and preventing regression!